src/main/java/org/opentripplanner/util/HttpUtils.java

```diff
@@ -13,40 +13,48 @@
 
 package org.opentripplanner.util;
 
-import java.io.FileNotFoundException;
-import java.io.IOException;
-import java.io.InputStream;
-
 import org.apache.http.HttpEntity;
 import org.apache.http.HttpResponse;
 import org.apache.http.StatusLine;
-import org.apache.http.client.ClientProtocolException;
 import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
 import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpHead;
-import org.apache.http.impl.client.DefaultHttpClient;
-import org.apache.http.params.BasicHttpParams;
-import org.apache.http.params.HttpConnectionParams;
-import org.apache.http.params.HttpParams;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
 
 public class HttpUtils {
-    
+
     private static final int TIMEOUT_CONNECTION = 5000;
+    private static final int TIMEOUT_CONNECTION_REQUEST = 5000;
     private static final int TIMEOUT_SOCKET = 5000;
 
     public static InputStream getData(String url) throws IOException {
-        return getData(url, null, null);
+        return getData(url, null, null, TIMEOUT_CONNECTION, TIMEOUT_CONNECTION_REQUEST, TIMEOUT_SOCKET);
     }
 
-    public static InputStream getData(String url, String requestHeaderName, String requestHeaderValue) throws ClientProtocolException, IOException {
+    public static InputStream getData(String url, int connectionTimeout, int connectionRequestTimeout, int socketTimeout) throws IOException {
+        return getData(url, null, null, connectionTimeout, connectionRequestTimeout, socketTimeout);
+    }
+
+    public static InputStream getData(String url, String requestHeaderName, String requestHeaderValue) throws IOException {
+        return getData(url, requestHeaderName, requestHeaderValue, TIMEOUT_CONNECTION, TIMEOUT_CONNECTION_REQUEST, TIMEOUT_SOCKET);
+    }
+
+    public static InputStream getData(String url, String requestHeaderName, String requestHeaderValue, int connectionTimeout, int connectionRequestTimeout, int socketTimeout) throws IOException {
```

Comment:
I generally like to avoid method signatures with more than 3 positional parameters. Do we really need three independent timeout settings? Is this method with five positional parameters actually used anywhere other than its wrapper methods? If you actually need to set all these timeouts independently to different values, can we somehow use non-static methods that read the configuration fields?